IntegrationTests/TimersTest.js

```diff
@@ -111,62 +119,77 @@ const TimersTest = createReactClass({
       0,
     );
     let failAnim; // This should fail without the t=0 fastpath feature.
-    this.setTimeout(() => this.cancelAnimationFrame(failAnim), 0);
-    failAnim = this.requestAnimationFrame(() =>
+    this._timeoutID = setTimeout(() => cancelAnimationFrame(failAnim), 0);
+    failAnim = requestAnimationFrame(() =>
       this._fail(
         'testOrdering-Anim, setTimeout 0 should happen before ' +
           'requestAnimationFrame',
       ),
     );
     let fail25;
-    this.setTimeout(() => {
-      this.clearTimeout(fail25);
+    this._timeoutID = setTimeout(() => {
+      clearTimeout(fail25);
     }, 20);
-    fail25 = this.setTimeout(
+    fail25 = setTimeout(
       () =>
         this._fail(
           'testOrdering-t25, setTimeout 20 should happen before ' +
             'setTimeout 25',
         ),
       25,
     );
-    this.setTimeout(this.done, 50);
-  },
+    this._timeoutID = setTimeout(this.done, 50);
+  }
 
   done() {
     this.setState({done: true}, () => {
       TestModule.markTestCompleted();
     });
-  },
+  }
+
+  componentWillUnmount() {
+    if (this._timeoutID != null) {
+      clearTimeout(this._timeoutID);
+    }
+    if (this._immediateId != null) {
+      clearImmediate(this._immediateId);
+    }
+    if (this._intervalId != null) {
+      clearInterval(this._intervalId);
+    }
+  }
 
   render() {
     return (
       <View style={styles.container}>
         <Text>
-          {this.constructor.displayName + ': \n'}
+          {/* $FlowFixMe(>=0.54.0 site=react_native_fb,react_native_oss) This
+             * comment suppresses an error found when Flow v0.54 was deployed.
+             * To see the error delete this comment and run Flow. */
+          this.constructor.displayName + ': \n'}
```

Comment:
IIRC, the displayName is added on to our components in a babel transform. Flow doesn't know anything about that, so that's probably why it's raising this error. 

I think there're three fixes:
1. Change `this.constructor.displayName` to `this.constructor.name`. (Both should be equivalent).
2. Redundantly define the `displayName` static property in this class. 
3. Hardcode 'TimerTest' instead of `this.constructor.displayName`. 

Your call. 😄